packages/fxa-auth-server/lib/db.js

```diff
@@ -322,26 +291,19 @@ module.exports = (config, log, Token, UnblockCode = null) => {
           .then(() => sessionTokens);
       }),
     ];
-    let isRedisOk = true;
 
     if (this.redis) {
-      promises.push(
-        this.safeRedisGet(uid).then(result => {
-          if (result === false) {
-            // Ensure that we don't return lastAccessTime if redis is down
-            isRedisOk = false;
-          }
-          return this.safeUnpackTokensFromRedis(uid, result);
-        })
-      );
+      promises.push(this.redis.getSessionTokens(uid));
     }
 
     return P.all(promises).spread(
       (mysqlSessionTokens, redisSessionTokens = {}) => {
+        // console.error(uid, redisSessionTokens)
         // for each db session token, if there is a matching redis token
         // overwrite the properties of the db token with the redis token values
-        const lastAccessTimeEnabled =
-          isRedisOk && features.isLastAccessTimeEnabledForUser(uid);
+        const lastAccessTimeEnabled = features.isLastAccessTimeEnabledForUser(
```

Comment:
good question. I don't know yet, but figuring it out is still on my todo list

Reply:
I've verified that when redis is down we return an empty tokens object which once we get down to `mergeDeviceInfoFromRedis` is ok because it doesn't override anything